### ISSUE TITLE
Add custom torch.nn loss to the MM module

### ISSUE THREAD

**Issue by KirderfEman** [autogluon/autogluon#2796]
Can you add simple setting in the MM predictor so the user can add its custom torch loss directly in the predictor.
One can do it manually as below, the code in the # are my functions, as example:

multimodal/optimization/utils.py

    if problem_type in [BINARY, MULTICLASS]:
        if mixup_active:
            loss_func = SoftTargetCrossEntropy()
        else:
            loss_func = nn.CrossEntropyLoss()
           # weights = torch.tensor([0.2, 0.8]).cuda()
           # loss_func = nn.CrossEntropyLoss(weight=weights)
           # loss_func = FocalLoss()
       #     loss_func = BiTemperedLogisticLoss(t1=0.7, t2=1.3, smoothing=0.1)
    elif problem_type == REGRESSION:
        if loss_func_name is not None:
            if "bcewithlogitsloss" in loss_func_name.lower():
                loss_func = nn.BCEWithLogitsLoss()
            else:
                loss_func = nn.MSELoss()
        else:
            loss_func = nn.MSELoss()
    elif problem_type == NER:
        loss_func = nn.CrossEntropyLoss(ignore_index=0)
    elif problem_type == OBJECT_DETECTION:
        return None
    elif problem_type is None:
        return None
    else:
        raise NotImplementedError

    return loss_func

There is a setting today called loss_function seed in config.yaml, you maybe will use that, I have tried but it doesn't work.

**Comment by sxjscience** [comment]
Yeah, we had the discussion around customized loss function (the purpose of adding the key: https://github.com/autogluon/autogluon/blob/22ea482e52067b4958f897b4628079249bc4f583/multimodal/src/autogluon/multimodal/configs/optimization/adamw.yaml#L37). (Added by @Linuxdex )

May be we should add a registry about loss functions.

**Comment by KirderfEman** [comment]
Yes it should be easy implemented, as my examples above. It's more clean doing it in the predictor settings than manually in the code.
